script.js

```diff
@@ -1,4 +1,246 @@
+
+var makeDeck = function () {
+  // Initialise an empty deck array
+  var cards = [];
+  // Initialise an array of the 4 suits in our deck. We will loop over this array.
+  var suits = ["hearts", "diamonds", "clubs", "spades"];
+
+  // Loop over the suits array
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    // Store the current suit in a variable
+    var currentSuit = suits[suitIndex];
+
+    // Loop from 1 to 13 to create all cards for a given suit
+    // Notice rankCounter starts at 1 and not 0, and ends at 13 and not 12.
+    // This is an example of a loop without an array.
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      // By default, the card name is the same as rankCounter
+      var cardName = rankCounter;
+
+      // If rank is 1, 11, 12, or 13, set cardName to the ace or face card's name
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+
+      // Add the new card to the deck
+      cards.push(card);
+
+      // Increment rank Counter to iterate over the next rank
+      rankCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+
+  // Return the completed card deck
+  return cards;
 };
+
+var makeDeckDone = makeDeck();
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cards array
+function shuffleDeck(cards) {
+  // Loop over the card deck array once
+  var index = 0;
+  while (index < cards.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cards.length);
+
+    // Select the card that corresponds to randomIndex
+    var randomItem = cards[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentItem = cards[index];
+    // Swap positions of randomCard and currentCard in the deck
+    cards[index] = randomItem;
+
+    cards[randomIndex] = currentItem;
+    // Increment currentIndex
+    index += 1;
+  }
+  // Return the shuffled deck
+  return cards;
+}
+
+// Function to Create NEW DECK
+
+function createNewDeck() {
+  var shuffleDeckDone = shuffleDeck(makeDeckDone);
+  return shuffleDeckDone;
+}
+
+//////// got blackjack?
+
+function gotBlackjack(handArray) {
+  var playerCardOne = handArray[0];
+  var playerCardTwo = handArray[1];
+
+  var gotBlackjack = false;
+
+  if (
+    (playerCardOne.name == "ace" && playerCardTwo.rank >= 10) ||
+    (playerCardOne.rank >= 10 && playerCardTwo.name == "ace")
+  ) {
+    gotBlackjack = true;
+  } else {
```

Comment:
You can technically omit this else statement since your default value for `gotBlackjack` is already false